api/app/signals/apps/email_integrations/utils.py

```diff
@@ -15,3 +24,76 @@ def _create_feedback_and_mail_context(signal: Signal):
         'negative_feedback_url': negative_feedback_url,
         'positive_feedback_url': positive_feedback_url,
     }
+
+
+def make_email_context(signal: Signal, additional_context: Optional[dict] = None) -> dict:
+    """
+    Makes a context dictionary containing all values needed for the email templates
+    Can add additional context, but will make sure that none of the default values are overridden.
+
+    For backwards compatibility the Signal and the Status are still added to the context
+    """
+
+    # Pattern used to filter links from the Signal text and text_extra
+    url_pattern = 'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'
```

Comment:
Will not catch ftp links. In general I think we need to look into this issue in a bit more detail. The Django URLValidator, for instance, is quite a but more complicated. Internationalized domains will match (see the unicode range in the Django URLValidator) etc. and thus will not be removed. (I tested locally.)

Although better than what we have right now we need to be more strict here for security reasons. Whether we block this PR until that issue is addressed it up for discussion.

Reply:
PS

```python
import re

url_pattern = 'http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+'

TEST_URIS = [
    'blah https://ü.invalid blah',
    'ftp://blah.com',
]

for case in TEST_URIS:
    print(re.sub(url_pattern, '', case))
```

These at least are not removed, I am sure there are more.

Reply:
Added more tests and changed the pattern to match the URLValidator pattern of Django, needed to make 2 small changes to that.